resources/pdns_recursor_service_systemd.rb

```diff
@@ -0,0 +1,68 @@
+#
+# Cookbook Name:: pdns
+# Resources:: pdns_recursor_service
+#
+# Copyright 2017, Aetrion, LLC DBA DNSimple
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+resource_name :pdns_recursor_service_systemd
+
+provides :pdns_recursor_service, os: 'linux' do |_node|
+  Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
+end
+
+action_class.class_eval do
+  def whyrun_supported?
```

Comment:
This feature enable the '--why-run' and we need this for evaluating that the resource will be updated.